enum.c

```diff
@@ -1385,6 +1393,188 @@ sort_by_cmp(const void *ap, const void *bp, void *data)
     return OPTIMIZED_CMP(a, b);
 }
 
+
+/*
+    This is parts of uniform sort
+*/
+
+#define ucmp rb_uniform_num_cmp_bool
+
+static inline bool
+rb_uniform_num_cmp_bool(VALUE a, VALUE b)
+{
+
+    return ((FIXNUM_P(a) && FIXNUM_P(b)) ?
+            ((long)a < (long)b) :
+            (RB_FLOAT_TYPE_P(a)) ?
+            (rb_float_cmp(a, b) == -1) :
+            (FIXNUM_P(a) && RB_FLOAT_TYPE_P(b))?
+            (rb_float_cmp(b, a) == 1) :
+            (RB_FLOAT_TYPE_P(a) && FIXNUM_P(b))?
+            (rb_float_cmp(a, b) == -1) :
+            (rb_cmpint(rb_funcallv(a, id_cmp, 1, &b), a, b)) == -1);
```

Comment:
This chain of conditional operators seems too complex.
`if`..`else if` feels more readable.

```suggestion
    if (FIXNUM_P(a) && FIXNUM_P(b)) {
        return (SIGNED_VALUE)a < (SIGNED_VALUE)b;
    }
    else if (RB_FLOAT_TYPE_P(a)) {
        return rb_float_cmp(a, b) == -1;
    }
    else if (FIXNUM_P(a) && RB_FLOAT_TYPE_P(b)) {
        return rb_float_cmp(b, a) == 1;
    }
    else if (RB_FLOAT_TYPE_P(a) && FIXNUM_P(b)) {
        return rb_float_cmp(a, b) == -1;
    }
    else {
        return rb_cmpint(rb_funcallv(a, id_cmp, 1, &b), a, b) == -1;
    }
```

And this function name doesn't feel matching with the behavior which returns `a is less than b?`, while other comparison functions such as `sort_by_cmp` return negative or positive value, or 0.